### PR TITLE
Handle network errors in proper way

### LINKED ISSUE #59: In case of network error it may happend that bubuku won't start
If unhandled exception will occur in run_daemon_loop cycle, that bubuku process will try to restart it's own work cycle, and recreate all internal entities (having in mind existing kafka process).
But if in that case AmazonEnvProvder.get_region() will return null, than it won't be able to refresh exhibitor address list, and that will start unlimited cycle of exceptions (see AWSExhibitorAddressProvider.get_addresses_by_lb_name).

### PR DESCRIPTION
**Pull request by antban**
Or at least allow to recover after it. Fixes #59

### CODE REVIEW

File: bubuku/env_provider.py [hunk 1]
```diff
@@ -35,46 +36,59 @@ def create_env_provider(config: Config):
...
-                _LOG.info("Amazon specific information loaded from AWS: {}".format(
-                    json.dumps(self.document, indent=2)))
-            except Exception as ex:
-                _LOG.warn('Failed to download AWS document', exc_info=ex)
-        return self.document
-
-    def get_aws_region(self) -> str:
-        doc = self._get_document()
-        return doc['region'] if doc else None
+        document = requests.get('http://{}/latest/dynamic/instance-identity/document'.format(self.aws_addr),
```
Comment (v-stepanov) [inline bubuku/env_provider.py]: What is this magic endpoint?
And what is the magic address '169.254.169.254'?

File: bubuku/env_provider.py [hunk 2]
```diff
@@ -35,46 +36,59 @@ def create_env_provider(config: Config):
...
 
     def _get_document(self) -> dict:
-        if not self.document:
-            try:
-                self.document = requests.get(
-                    'http://{}/latest/dynamic/instance-identity/document'.format(self.aws_addr),
-                    timeout=5).json()
-                _LOG.info("Amazon specific information loaded from AWS: {}".format(
-                    json.dumps(self.document, indent=2)))
-            except Exception as ex:
```
Comment (v-stepanov) [inline bubuku/env_provider.py]: Why you removed exception handling?

File: bubuku/env_provider.py (hunk 1 above)
Comment (antban) [inline bubuku/env_provider.py]: It's not magic :)
http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html

File: bubuku/env_provider.py (hunk 2 above)
Comment (antban) [inline bubuku/env_provider.py]: Because for "Amazon" class it's really exception when it's not possible to connect to AWS

**Comment by antban** [comment]
@v-stepanov @rcillo @adyach @v1ctor  Are there any other suggestions? This bugfix is really important.

**Comment by retnuh** [comment]
@antban Hey any chance of a new release with this fix?  We're using bubuku ourselves, and seen some strange problems, and would like to know if the recent fixes clear the problems up.

Thanks,

H

**Comment by antban** [comment]
@retnuh Wow, lost this comment. I plan to release on Monday evening (19.09) or 20.09. The thing is that I want to include optimized rebalancing. It's one of the important features.

**Pull request merged**